### PR TITLE
Updated Dsc resource example

### PR DESCRIPTION
**Pull request by bgelens**
* added schema generation
*  added IsSingleInstance inquiry 
* added Reboot required inquiry
* fixed function names to be just Get, Set, Test-TargetResource

### CODE REVIEW

**Comment by daviwil** [comment]
Ben is going to work on an uber template for creating DSC resources :)  I'm excited!

File: examples/NewDscResourceScript/resource.ps1.txt [hunk 1]
```diff
@@ -1,57 +1,104 @@
...
     if ($PLASTER_PARAM_Ensure -eq 'Yes') {
 "        # Ensure the presence/absene of the resource."
+"        [Parameter(Mandatory)]"
 "        [ValidateSet('Present','Absent')]"
-"        [string]"
-"        `$Ensure = 'Present'"
+         if ($PLASTER_PARAM_IsSingleInstance -eq 'Yes') {
+"        [string] `$Ensure,"
+""
+         } else {
```
Comment (rkeithhill) [inline examples/NewDscResourceScript/resource.ps1.txt]: Could you tweak this?  Nowhere else in this code base do we use a cuddled else.  Please move the `else {` to a new line.

Comment (bgelens) [inline examples/NewDscResourceScript/resource.ps1.txt]: habits :)

**Comment by rkeithhill** [comment]
@bgelens  Thanks!

**Pull request merged**